File: src/main/scala/im/tox/toktok/app/SimpleDialogs/SimpleDialogDesign.scala
package im.tox.toktok.app.SimpleDialogs

import android.app.{Activity, Dialog}
import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.View.OnClickListener
import android.view.animation.AnimationUtils
import android.view.{View, Window}
import android.widget._
import im.tox.toktok.R

class SimpleDialogDesign(activity: Activity, question: String, color: Int, icon: Int, clickAction: OnClickListener) extends Dialog(activity, R.style.DialogSlideAnimation) {

  var a: Activity = activity
  var mBase : RelativeLayout = null

  override def onCreate(savedInstanceState: Bundle): Unit = {
    super.onCreate(savedInstanceState)
    requestWindowFeature(Window.FEATURE_NO_TITLE)
    setContentView(R.layout.simple_dialog_design)
    getWindow.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT))

    findViewById(R.id.simple_dialog_color).asInstanceOf[LinearLayout].setBackgroundTintList(ColorStateList.valueOf(color))
    findViewById(R.id.simple_dialog_img).asInstanceOf[ImageView].setImageResource(icon)

    findViewById(R.id.simple_dialog_text).asInstanceOf[TextView].setText(question)

    mBase = findViewById(R.id.simple_dialog_base).asInstanceOf[RelativeLayout]

    val confirmButton = findViewById(R.id.simple_dialog_confirm).asInstanceOf[Button]
    confirmButton.setOnClickListener(clickAction)


    val cancelButton = findViewById(R.id.simple_dialog_cancel).asInstanceOf[Button]
    cancelButton.setOnClickListener(new OnClickListener {
      override def onClick(v: View): Unit = {
        dismiss()
      }
    })

  }

}

File: sources/app/src/main/java/im/tox/toktok/app/simple_dialogs/SimpleDialogDesign.java
package im.tox.toktok.app.simple_dialogs;

import android.app.Activity;
import android.app.Dialog;
import android.content.res.ColorStateList;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import androidx.annotation.NonNull;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import im.tox.toktok.R;

public final class SimpleDialogDesign extends Dialog {

    private final String question;
    private final int color;
    private final int icon;
    private final View.OnClickListener clickAction;

    public SimpleDialogDesign(
            @NonNull Activity activity,
            String question,
            int color,
            int icon,
            View.OnClickListener clickAction
    ) {
        super(activity, R.style.DialogSlideAnimation);
        this.question = question;
        this.color = color;
        this.icon = icon;
        this.clickAction = clickAction;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.simple_dialog_design);
        getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));

        this.findViewById(R.id.simple_dialog_color).setBackgroundTintList(ColorStateList.valueOf(color));
        this.<ImageView>findViewById(R.id.simple_dialog_img).setImageResource(icon);
        this.<TextView>findViewById(R.id.simple_dialog_text).setText(question);

        Button confirmButton = this.findViewById(R.id.simple_dialog_confirm);
        confirmButton.setOnClickListener(clickAction);

        Button cancelButton = this.findViewById(R.id.simple_dialog_cancel);
        cancelButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
            }
        });
    }

}
